api/gateway.go

```diff
@@ -50,30 +48,15 @@ func NewGateway(n *core.OpenBazaarNode, authCookie http.Cookie, l net.Listener,
 	}
 
 	return &Gateway{
-		listener:   l,
-		handler:    topMux,
-		config:     config,
-		shutdownCh: make(chan struct{}),
+		listener: l,
+		handler:  topMux,
+		config:   config,
 	}, nil
 }
 
 // Close shutsdown the Gateway listener
 func (g *Gateway) Close() error {
 	log.Infof("server at %s terminating...", g.listener.Addr())
-
-	// Print shutdown message every few seconds if we're taking too long
-	go func() {
-		select {
-		case <-g.shutdownCh:
-			return
-		case <-time.After(5 * time.Second):
-			log.Infof("waiting for server at %s to terminate...", g.listener.Addr())
-
-		}
-	}()
-
-	// Shutdown the listener
-	close(g.shutdownCh)
```

Comment:
What was this protecting that it is no longer needed?

Reply:
Looks like this code does nothing to me. Amirite?

Reply:
That's what I'm confused about it. It's really not obvious to me why it was there.

Reply:
It appears that the goroutine with the select was supposed to have a loop around the select, so that it would log every few seconds until is g.shutdownCh is closed. This is a good idiom if the the shutdown occurs somewhere else, but in this case the goroutine is set up and then the channel is closed immediately. The message will never fire in practice and even if it did it would only print once and quit instead of printing until shutdown.

The code never worked as intended and just made debugging test issues I ran into harder to debug because it caused me to focus on a spot in the code that didn't matter.